tokio-threadpool/src/pool/mod.rs

```diff
@@ -550,7 +548,7 @@ impl Pool {
                 *rng = Some(new_rng(thread_id));
             }
 
-            rng.as_mut().unwrap().next_u32() as usize
+            rng.as_mut().unwrap().gen::<u32>() as usize
```

Comment:
BTW, it would be better to gen usize instead of casting u32 to usize on 64-bit machines.